app/js/controllers/playerListCtrl.js

```diff
@@ -2,8 +2,43 @@
 
 /* Controllers */
 
-angular.module('KMCModule').controller('PlayerListCtrl',
-	['apiService', 'loadINI', '$location', '$rootScope', '$scope', '$filter', '$modal', '$timeout', '$log', "$compile", "$window", 'localStorageService', 'requestNotificationChannel', 'PlayerService', '$q', 'utilsSvc', 'PermissionsService',
+angular.module('KMCModule')
+	.controller('PlayerUpgradeModeCtrl',
+		function ($scope, $modalInstance, settings) {
+			$scope.playerId = '';
+			$scope.mode = '';
+
+			$scope.close = function (result) {
+				var modalResult = result ? {
+					result: true,
+					mode: this.mode,
+					templateId: Number(this.playerId)
+				} : {
+					result: false
+				};
+				$modalInstance.close(modalResult);
```

Comment:
must you send `{ result: false }` as **no result**? can't you just send `{}` or something empty which represents **no result**.
```suggestion
				if(!result) {
				  return $modalInstance.close();
				}
				
				$modalInstance.close({
				  mode: this.mode,
				  templateId: this.playerId
				});
```

Reply:
why do you need the **Number** value from the `playerId`?
are you about to make number manipulations somehow with the value?
if not, let's make it simple and keep it like `templateId: this.playerId;`

Reply:
PlayerID is a UiConf ID, so it should be numeric. `Number` is used as mechanism to trim user input value.

Reply:
> must you send `{ result: false }` as **no result**? can't you just send `{}` or something empty which represents **no result**.

Yes, this will work.